bokehjs/src/lib/models/annotations/box_annotation.ts

```diff
@@ -72,24 +87,160 @@ export class BoxAnnotationView extends AnnotationView {
   }
 
   override interactive_hit(sx: number, sy: number): boolean {
-    if (this.model.in_cursor == null)
+    if (!this.model.visible || !this.model.editable)
       return false
     const bbox = this.interactive_bbox()
     return bbox.contains(sx, sy)
   }
 
-  override cursor(sx: number, sy: number): string | null {
-    const tol = 3
-
+  private _hit_test(sx: number, sy: number): HitTarget | null {
     const {left, right, bottom, top} = this.bbox
-    if (Math.abs(sx-left) < tol || Math.abs(sx-right) < tol)
-      return this.model.ew_cursor
-    else if (Math.abs(sy-bottom) < tol || Math.abs(sy-top) < tol)
-      return this.model.ns_cursor
-    else if (this.bbox.contains(sx, sy))
-      return this.model.in_cursor
-    else
-      return null
+    const tolerance = Math.max(EDGE_TOLERANCE, this.model.line_width/2)
+
+    const hits_left = abs(left - sx) < tolerance
+    const hits_right = abs(right - sx) < tolerance
```

Comment:
If the box is really narrow, this can result in both `hits_left` and `hits_right` being `true` and subsequently the left is preferred, even if the cursor is just to the right of both edges. I think we can do better by only allowing one of the two to be set, the edge that is nearest to the cursor position. This can be achieved by adding a subsequent check something like (untested):
```typescript
if (hits_left && hits_right) {
  if (abs(left - sx) < abs(right - sx))
    hits_right = false
  else
    hits_left = false
}
```
or alternatively finding the nearest edge first and only checking the tolerance for that edge.

And the same for `hits_top` and `hits_bottom` too.